packages/phishing-controller/src/PhishingDetector.ts

```diff
@@ -0,0 +1,177 @@
+import { distance } from 'fastest-levenshtein';
+
+import {
+  domainPartsToDomain,
+  domainPartsToFuzzyForm,
+  domainToParts,
+  getDefaultPhishingDetectorConfig,
+  matchPartsAgainstList,
+  processConfigs,
+} from './utils';
+
+export type LegacyPhishingDetectorList = {
+  whitelist?: string[];
+  blacklist?: string[];
+} & FuzzyTolerance;
+
+export type PhishingDetectorList = {
+  allowlist?: string[];
+  blocklist?: string[];
+  name?: string;
+  version?: string | number;
+} & FuzzyTolerance;
+
+export type FuzzyTolerance =
+  | {
+      tolerance?: number;
+      fuzzylist: string[];
+    }
+  | {
+      tolerance?: never;
+      fuzzylist?: never;
+    };
+
+export type PhishingDetectorOptions =
+  | LegacyPhishingDetectorList
+  | PhishingDetectorList[];
+
+export type PhishingDetectorConfiguration = {
+  name?: string;
+  version?: number | string;
+  allowlist: string[][];
+  blocklist: string[][];
+  fuzzylist: string[][];
+  tolerance: number;
+};
```

Comment:
This might be confusing, but the main difference between the options passed to the constructor, and the configs saved in `#configs` (`PhishingDetectorConfiguration`) is that domains are divided in parts, so each list becomes a matrix